### DISCUSSION TITLE
Plugin is not working if fingerprint is not available in device.

### DISCUSSION THREAD

**Discussion by siraj-coder** [NiklasMerz/cordova-plugin-fingerprint-aio#447] [Q&A]
When we try to integrate AIO plugin with our app it is giving error as **hardware not supported** , as my device does not support fingerprint but it supports face id and pin. Can we integrate this plugin with my app as only face id or pin authentication?

Currently my app is built on :

Android version: 11
Ionic version: 5.36.x
Angular version: 12.2.x

**Comment by NiklasMerz** [comment]
The plugin is checking Androids BiometricManager if it has hardware support. The plugin relies on BiometricManager to do all the authentication work. I am not sure if you could make it work with no hardware support.

https://github.com/NiklasMerz/cordova-plugin-fingerprint-aio/blob/175549c4ab1fe45b98c9fb1556cb523f39eac5cf/src/android/Fingerprint.java#L141-L143

**You might need to implement a fallback PIN logic in your app.**
